File: experiments/issue-301/probe.py
import sys

sys.path.insert(0, "python")
from links_notation.parser import Parser

docs = ["# a b\n", "# a: b\n", "a: b # note\n", "a#b\n", '"#" a\n', "parent\n  # what the child is for\n  child\n"]
for doc in docs:
    try:
        links = Parser().parse(doc)
        print(f"{doc!r} -> PARSED [{' '.join(str(link) for link in links)}]")
    except Exception as error:  # noqa: BLE001 - the probe reports whatever comes out
        print(f"{doc!r} -> {type(error).__name__}: {error}")

File: experiments/issue-301/probe.php
<?php
spl_autoload_register(function ($class) {
    $prefix = 'LinkFoundation\\LinksNotation\\';
    if (str_starts_with($class, $prefix)) {
        require __DIR__ . '/../../php/src/' . str_replace('\\', '/', substr($class, strlen($prefix))) . '.php';
    }
});
$docs = ["# a b\n", "# a: b\n", "a: b # note\n", "a#b\n", '"#" a' . "\n", "parent\n  # what the child is for\n  child\n"];
foreach ($docs as $doc) {
    try {
        $links = (new LinkFoundation\LinksNotation\Parser())->parse($doc);
        echo json_encode($doc) . ' -> PARSED [' . implode(' ', array_map(fn ($link) => (string) $link, $links)) . "]\n";
    } catch (Throwable $error) {
        echo json_encode($doc) . ' -> ' . get_class($error) . ': ' . $error->getMessage() . "\n";
    }
}
